### PR TITLE
MAINT: Work with matplotlib 3.6 dep

### PR DESCRIPTION
**Pull request by larsoner**
### Overview
matplotlib 3.6 deprecated `get_cmap` in favor of `matplotlib.colormaps[string_name]`.

On my macOS M1 system VTK appears not to have vtkRedistributeDataSetFilter. Probably a VTK problem but let's at least keep pyvista usable on these systems for now.

### CODE REVIEW

**Comment by adeak** [comment]
> On my macOS M1 system VTK appears not to have vtkRedistributeDataSetFilter. Probably a VTK problem but let's at least keep pyvista usable on these systems for now.

@whophil ran into this too recently. Seems to be a shortcoming of the conda-forge vtk package: the pypi wheel works fine.

**Comment by whophil** [comment]
> On my macOS M1 system VTK appears not to have vtkRedistributeDataSetFilter. Probably a VTK problem but let's at least keep pyvista usable on these systems for now.

I agree with the choice to make this a non-required import. Nevertheless, the VTK conda package will hopefully be repaired soon, see https://github.com/conda-forge/vtk-feedstock/pull/252

File: pyvista/_vtk.py [hunk 1]
```diff
@@ -263,7 +263,11 @@
         vtkTrimmedExtrusionFilter,
     )
     from vtkmodules.vtkFiltersParallel import vtkIntegrateAttributes
-    from vtkmodules.vtkFiltersParallelDIY2 import vtkRedistributeDataSetFilter
+
+    try:
+        from vtkmodules.vtkFiltersParallelDIY2 import vtkRedistributeDataSetFilter
+    except ModuleNotFoundError:  # macOS M1 VTK 9.2.0rc1
+        pass
```
Comment (akaszynski) [inline pyvista/_vtk.py:270]:
```suggestion
    except ModuleNotFoundError:  # pragma: no cover
        # unavailable on macOS M1 VTK 9.2.0rc1
        pass
```

**Review by akaszynski — APPROVED** [review]

File: pyvista/_vtk.py [hunk 2]
```diff
@@ -263,7 +263,11 @@
         vtkTrimmedExtrusionFilter,
     )
     from vtkmodules.vtkFiltersParallel import vtkIntegrateAttributes
-    from vtkmodules.vtkFiltersParallelDIY2 import vtkRedistributeDataSetFilter
+
+    try:
+        from vtkmodules.vtkFiltersParallelDIY2 import vtkRedistributeDataSetFilter
+    except ModuleNotFoundError:  # pragma: no cover
+        pass  # unavailable on macOS M1 VTK 9.2.0rc1
```
Comment (whophil) [inline pyvista/_vtk.py:270]: I think this is not specifically a MacOS M1 thing, but rather an issue with the VTK conda package. Minor detail, but the comment could lead to some confusion in the future. @larsoner did you try this import with a pip-installed VTK 9.2.0 wheel on your M1 machine?

Comment (adeak) [reply pyvista/_vtk.py:270]: We also covered in slack that it's missing for 9.0 and 9.1 conda-forge VTK.

Comment (whophil) [reply pyvista/_vtk.py:270]: For reference: https://pyvista.slack.com/archives/CBA5095UH/p1661028781796529

Comment (akaszynski) [reply pyvista/_vtk.py:270]: > I didn't notice that auto-merge was enabled...

My bad. Force of habit, been merging a ton elsewhere. @whophil, can you open up a follow-up?

**Pull request merged**